File: src/main/java/brs/grpc/handlers/ParseTransactionHandler.java
package brs.grpc.handlers;

import brs.DependencyProvider;
import brs.Transaction;
import brs.grpc.GrpcApiHandler;
import brs.grpc.proto.BrsApi;
import brs.grpc.proto.ProtoBuilder;

public class ParseTransactionHandler implements GrpcApiHandler<BrsApi.TransactionBytes, BrsApi.BasicTransaction> {
    private final DependencyProvider dp;

    public ParseTransactionHandler(DependencyProvider dp) {
        this.dp = dp;
    }

    @Override
    public BrsApi.BasicTransaction handleRequest(BrsApi.TransactionBytes transactionBytes) throws Exception {
        return ProtoBuilder.buildBasicTransaction(Transaction.parseTransaction(dp, transactionBytes.getTransactionBytes().toByteArray()));
    }
}

File: src/main/java/brs/grpc/handlers/ParseTransactionHandler.kt
package brs.grpc.handlers

import brs.DependencyProvider
import brs.Transaction
import brs.grpc.GrpcApiHandler
import brs.grpc.proto.BrsApi
import brs.grpc.proto.ProtoBuilder

class ParseTransactionHandler(private val dp: DependencyProvider) : GrpcApiHandler<BrsApi.TransactionBytes, BrsApi.BasicTransaction> {

    override suspend fun handleRequest(transactionBytes: BrsApi.TransactionBytes): BrsApi.BasicTransaction {
        return ProtoBuilder.buildBasicTransaction(Transaction.parseTransaction(dp, transactionBytes.transactionBytes.toByteArray()))
    }
}
